### PR TITLE
Update docs for focusIndicator prop on Select

### PR DESCRIPTION
**Pull request by markjackson02**
#### What does this PR do?
Update documentation to reflect current functionality. Current documentation indicates that `focusIndicator` prop is only applicable if `plain={true}`. However this is only true of the `TextInput`. For `Button` and `Select`, `focusIndicator` works without the prerequisite of `plain={true}`.

#### Where should the reviewer start / How should this be manually tested?
You can confirm that `focusIndicator` works without `plain` by testing adding `focusIndicator={false}` on the interactive demos.
https://v2.grommet.io/button
https://v2.grommet.io/select

#### What testing has been done on this PR?
N/A - only documentation updates

#### Should this PR be mentioned in the release notes?
I don't think so

### CODE REVIEW

File: src/js/components/Button/doc.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ export const doc = Button => {
       )
       .defaultValue(false),
     focusIndicator: PropTypes.bool
-      .description("Whether when 'plain' it should receive a focus outline.")
+      .description('Whether it should receive a focus outline.')
```
Comment (ShimiSun) [inline src/js/components/Button/doc.js:50]: I think this is only true for `plain` button.

Comment (markjackson02) [reply src/js/components/Button/doc.js:50]: Hmm you're right I must have tested on the wrong page. So it looks like Select is the only one whose docs need updating. Will update PR

Comment (ShimiSun) [reply src/js/components/Button/doc.js:50]: `  ${props =>
    props.focus && (!props.plain || props.focusIndicator) && focusStyle()}`

**Comment by ShimiSun** [comment]
I'll be closing this PR since the `focusIndicator` should apply only on `plain` inputs or controls.
Linking to this issue https://codesandbox.io/s/grommet-v2-template-xe041?file=/index.js

**Pull request closed without merging**